src/cascadia/TerminalApp/Pane.h

```diff
@@ -130,4 +140,24 @@ class Pane : public std::enable_shared_from_this<Pane>
         }
         return false;
     }
+
+    // Helper structure that builds a (roughly) binary tree corresponding
+    // to the pane tree. Used for layouting panes with snapped sizes.
+    struct LayoutSizeNode
+    {
+        float size;
+        bool isMinimumSize;
+        std::unique_ptr<LayoutSizeNode> firstChild;
+        std::unique_ptr<LayoutSizeNode> secondChild;
+        std::unique_ptr<LayoutSizeNode> nextFirstChild;

```

Comment:
I think that's mostly correct.
`firstChild` and `secondChild` are the `current`, snapped sizes (`current` in the progression of layouting algorithm). I'm not sure if "next smaller snap size" does describe that correctly.

`next*Child` are the next (higher) possible snapped sizes. In simplest case they represent the `*Child` with one more character. They serve mostly like a cache - we might calculate them each time `firstChild` and `secondChild` but this would be wasteful because  on each `AdvanceSnappedDimension` call we have to know them both but we choose only one. 

This also means that the bulk on 'advancing' say first child becomes `*firstChild = *nextFirstChild`, assuming `nextFirstChild` is already calculated (and then of course we have to adjust `nextFirstChild` itself).

Reply:
Okay, I think this should probably be added as a comment, because I certainly didn't feel like I understood that on my first read-through of the code.